packages/next-frontend/src/components/create-campaign/form/end-date-step/index.tsx

```diff
@@ -1,26 +1,163 @@
 import { useTranslations } from "next-intl";
+import { useChainId } from "wagmi";
+import { useCallback, useEffect, useState } from "react";
+import dayjs, { type Dayjs, type ManipulateType } from "dayjs";
 import { Step } from "@/src/components/step";
 import { StepPreview } from "@/src/components/step/preview";
 import { StepContent } from "@/src/components/step/content";
 import type { CampaignPayload, CampaignPayloadPart } from "@/src/types";
+import { Typography } from "@/src/ui/typography";
+import { DateTimePicker } from "@/src/ui/date-time-picker";
+import { Button } from "@/src/ui/button";
+import { Chip } from "@/src/ui/chip/chip";
+
+import styles from "./styles.module.css";
 
 interface EndDateStepProps {
     disabled?: boolean;
+    startDate?: CampaignPayload["endDate"];
     endDate?: CampaignPayload["endDate"];
-    onEndDateChange?: (startDate: CampaignPayloadPart) => void;
+    onEndDateChange: (startDate: CampaignPayloadPart) => void;
+}
+
+interface DurationPreset {
+    label: string;
+    unit: ManipulateType;
+    value: number;
 }
 
+const DURATION_PRESETS: DurationPreset[] = [
+    {
+        label: "durations.1week",
+        unit: "days",
+        value: 7,
+    },
+    {
+        label: "durations.2weeks",
+        unit: "days",
+        value: 14,
+    },
+    {
+        label: "durations.3weeks",
+        unit: "days",
+        value: 21,
+    },
+    {
+        label: "durations.1month",
+        unit: "months",
+        value: 1,
+    },
+];
+
 export function EndDateStep({
     disabled,
+    startDate,
     endDate,
     onEndDateChange,
 }: EndDateStepProps) {
     const t = useTranslations("new_campaign.form.end_date");
+    const [open, setOpen] = useState(false);
+    const [date, setDate] = useState<Dayjs | undefined>(endDate);
+    const [durationPreset, setDurationPreset] = useState<DurationPreset>();
+    const chainId = useChainId();
+
+    useEffect(() => {
+        setOpen(false);
+    }, [chainId]);
+
+    function handleStepOnClick() {
+        if (open && !endDate) setDate(undefined);
+        if (open) setDate(endDate);
+        setOpen((open) => !open);
+    }
+
+    function handleDateOnChange(value: Dayjs) {
```

Comment:
Shouldn't we pass `setDate` directly here?

Reply:
Fixed in 64b94f25b91cbb89a3677a6ed45bb3df5ea96415.